rs/registry/canister/src/registry_lifecycle.rs

```diff
@@ -275,4 +305,46 @@ mod test {
             );
         }
     }
+
+    #[test]
+    fn test_migrate_node_reward_type1_type0_to_type1dot1_works_correctly() {
+        let mut registry = invariant_compliant_registry(0);
+
+        let mut node_additions = Vec::new();
+        for (idx, test_id) in enumerate(0..10) {
+            let node_reward_type = if idx < 5 {
+                NodeRewardType::Type0
+            } else {
+                NodeRewardType::Type1
+            };
+            let record = NodeRecord {
+                node_operator_id: PrincipalId::new_anonymous().to_vec(),
+                node_reward_type: Some(node_reward_type as i32),
+                ..NodeRecord::default()
+            };
+
+            node_additions.push(insert(
+                make_node_record_key(NodeId::new(PrincipalId::new_node_test_id(test_id))),
+                record.encode_to_vec(),
+            ));
+        }
+
+        registry.apply_mutations_for_test(node_additions);
+        let mutations = migrate_node_reward_type1_type0_to_type1dot1(&registry);
+        assert_eq!(mutations.len(), 10);
+
+        registry.apply_mutations_for_test(mutations);
+
+        for test_id in 0..10 {
+            let record =
+                registry.get_node_or_panic(NodeId::from(PrincipalId::new_node_test_id(test_id)));
+
+            assert_eq!(
```

Comment:
You need another assertion to show that other fields that aren't targeted are unchanged.  Add some variable data in the record above.  

There isn't really data recovery if you do a bad migration.